### DISCUSSION TITLE
如果mysql cdc 监听多张表，现在想要重跑其中两张，可以做到吗？

### DISCUSSION THREAD

**Discussion by lliuql** [apache/flink-cdc#3184] [Q&A]
如果遇到新表的话，目前我使用的时动态加表的形式。scanNewlyAddedTableEnabled(true) 。但是如果要重跑本身已经有的表，有办法吗

**Comment by loserwang1024** [comment]
可以尝试先exclude掉，然后在include加进去

**Comment by wufanwind** [comment]
scanNewlyAddedTableEnabled(true) 能监听到mysql里新建的表的数据吗？ 我测试自从2.4版本以后都监听不到了，必须重启cdc任务才有。

**Reply by loserwang1024** [reply]
目前的设计就是只能重启才能读取新表。2.4初期版本，如果读取binlog, 符合正则表达式就会读取并且发送下游，但是没有在enumerator注册表数据。这会导致重启之后，enumerator将这些表当作新表，再重复消费。如果作业运行一周，就会导致一周的数据重复。因此从2.4版本动态加表只有重启之后才能读取新表。后续社区会修复该问题，在enumerator注册表数据。如果你有兴趣，也可以参与贡献。
